<script>
    const display=()=>{
        return new Promise((success,failure)=>{
            try{
                fetch('https://jsonplaceholder.typicode.com/todos')
      .then(response => { 
        return response.json()
      }
     )
      .then(json => {
        success(json);
      })
            }catch(err){
                failure("some error occured");
            }
          
        })
    }

    const subDisplay=(id)=>{
        return new Promise((success,failure)=>{
            try{
                fetch(`https://jsonplaceholder.typicode.com/todos/${id}`)
      .then(response => { 
        return response.json()
      }
     )
      .then(json => {
        success(json);
      })
            }catch(err){
                failure("some error occured");
            }
          
        })
    }

    display().then((result)=>{
     console.log("promise success",result);
     return subDisplay(result[0].id);
    },(error)=>{
        console.log("promise failed")
    }).then((sub)=>{
       console.log("second promise result",sub);
    })
</script>